administrator/components/com_admin/script.php

```diff
@@ -282,6 +285,67 @@ protected function uninstallExtensions()
         }
     }
 
+    /**
+     * This method is for migration for old logrotation system plugin migration to task.
+     *
+     * @param   \stdClass  $data  Object with `extension_id` and `params` of the extension
+     *
+     * @return  void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    private function migrateLogRotationPlugin($data)
+    {
+        /** @var SchedulerComponent $component */
+        $component = Factory::getApplication()->bootComponent('com_scheduler');
+
+        /** @var TaskModel $model */
+        $model = $component->getMVCFactory()->createModel('Task', 'Administrator', ['ignore_request' => true]);
+
+        if ($plugin = PluginHelper::getPlugin('system', 'actionlogs')) {
+            $pluginParams = new Registry($plugin->params);
+            $lastrun      = (int) $pluginParams->get('lastrun', 0);
+            $task = [
+                'title' => 'DeleteActionLogs',
+                'type'  => 'delete.actionlogs',
+                'execution_rules' => [
+                    'rule-type' => 'interval-hours',
+                    'interval-hours' => 24,
+                    'exec-time' => gmdate("H:i", $lastrun),
+                ],
+                'state' => 1,
+                'params' => [
+                    'logDeletePeriod' => $pluginParams->get('logDeletePeriod', 0),
+                ]
+            ];
+            $model->save($task);
+        }
+
+        if (!$data->enabled) {
+            return;
+        }
+
+        // Get the timeout, as configured in plg_system_logrotation
+        $params       = new Registry($data->params);
+        $cachetimeout = (int) $params->get('cachetimeout', 6);
+        $lastrun      = (int) $params->get('lastrun', 0);
+
+        $task = [
+            'title' => 'RotateLogs',
+            'type'  => 'rotation.logs',
+            'execution_rules' => [
+                'rule-type' => 'interval-hours',
+                'interval-hours' => $cachetimeout,
+                'exec-time' => gmdate("H:i", $lastrun),
+            ],
+            'state' => 1,
+            'params' => [
+                'logstokeep' => $params->get('logstokeep', 0),
+            ]
+        ];
```

Comment:
```suggestion
        $task = [
            'title'           => 'RotateLogs',
            'type'            => 'rotation.logs',
            'execution_rules' => [
                'rule-type'      => 'interval-hours',
                'interval-hours' => $cachetimeout,
                'exec-time'      => gmdate("H:i", $lastrun),
            ],
            'state'  => 1,
            'params' => [
                'logstokeep' => $params->get('logstokeep', 0),
            ],
        ];
```
Fix PHPCS (last one, I think).